src/NewsService.Business/CreateNewsCommand.cs

```diff
@@ -17,32 +25,74 @@ public class CreateNewsCommand : ICreateNewsCommand
         private readonly IDbNewsMapper _mapper;
         private readonly INewsValidator _validator;
         private readonly IAccessValidator _accessValidator;
+        private readonly IRequestClient<ICheckDepartmentsExistence> _rcCheckDepartmentsExistence;
+        private readonly ILogger<CreateNewsCommand> _logger;
+
+        private List<Guid> CheckDepartmentExistence(Guid? departmentId, List<string> errors)
+        {
+            string errorMessage = "Failed to check the existing department.";
+            string logMessage = "Department with id {id} not found.";
+
+            try
+            {
+                var response = _rcCheckDepartmentsExistence.GetResponse<IOperationResult<ICheckDepartmentsExistence>>(
+                    ICheckDepartmentsExistence.CreateObj(new List<Guid> { departmentId.Value })).Result;
+                if (response.Message.IsSuccess)
+                {
+                    if (!response.Message.Body.DepartmentIds.Any())
+                    {
+                        errors.Add($"Department Id: {departmentId} does not exist");
+                    }
+                    return response.Message.Body.DepartmentIds;
+                }
+
+                _logger.LogWarning($"Can not find {departmentId} with this Ids '{departmentId}': " +
+                    $"{Environment.NewLine}{string.Join('\n', response.Message.Errors)}");
+            }
+            catch (Exception exc)
+            {
+                _logger.LogError(exc, logMessage);
+            }
+
+            errors.Add(errorMessage);
+            return new List<Guid>();
```

Comment:
need return null, so as Body response should be null value in this case

Reply:
fixed